packages/kbn-mock-idp-plugin/server/plugin.ts

```diff
@@ -77,109 +80,291 @@ export type CreateSAMLResponseParams = TypeOf<typeof createSAMLResponseSchema>;
+          return response.redirected({ headers: { location: '/' } });
+        }
+      );
+
+      router.post(
+        {
+          path: '/mock_idp/grant_api_key',
+          validate: {
+            body: schema.object({
+              name: schema.string(),
+              expiration: schema.maybe(schema.string()),
+              authcScheme: schema.maybe(schema.string()),
+              credential: schema.maybe(schema.string()),
+            }),
+          },
+          security: { authz: { enabled: false, reason: 'Mock IDP plugin for testing' } },
+        },
+        async (context, request, response) => {
+          try {
+            if (!getAuthenticationService) {
+              return response.badRequest({
+                body: { message: 'Authentication service not available yet' },
+              });
+            }
+
+            const { name, authcScheme, credential, expiration } = request.body;
+            const authcService = getAuthenticationService();
+
+            // Create a new request with authentication header if authcScheme and credential are provided
+            let requestToUse = request;
+            if (authcScheme && credential) {
+              const requestHeaders: Headers = {
+                ...request.headers,
+                authorization: `${authcScheme} ${credential}`,
+              };
+              const fakeRawRequest: FakeRawRequest = {
+                headers: requestHeaders,
+                path: request.url.pathname,
+              };
+              requestToUse = kibanaRequestFactory(fakeRawRequest);
+            }
+
+            const result = await authcService.apiKeys.uiam.grantApiKey(requestToUse, {
+              name,
+              expiration,
+            });
+
+            if (!result) {
+              return response.badRequest({
+                body: { message: 'Failed to grant API key' },
+              });
+            }
+
+            return response.ok({
+              body: result,
+            });
+          } catch (err) {
+            logger.error(`Failed to grant API key: ${err}`, err);
+            return response.customError({
+              statusCode: 500,
+              body: { message: err.message },
+            });
           }
-          return response.ok({
-            body: {
-              roles,
-            },
-          });
-        } catch (err) {
-          return response.customError({ statusCode: 500, body: err.message });
         }
-      }
-    );
-
-    router.post(
-      {
-        path: '/mock_idp/saml_response',
-        validate: {
-          body: createSAMLResponseSchema,
+      );
+
+      router.post(
+        {
+          path: '/mock_idp/call_scoped_client_with_uiam_api_key',
+          validate: {
+            body: schema.object({
+              apiKey: schema.string(),
+              authcScheme: schema.string(),
+            }),
+          },
+          options: { authRequired: 'optional' },
```

Comment:
__nit:__ here and below, can we use non-deprecated `security: { authc: { enabled: 'optional' },` instead?